src/api/starDashboard.js

```diff
@@ -0,0 +1,17 @@
+const getStarDashboardMutation = id => ({
+    type: 'create',
+    resource: `dashboards/${id}/favorite`,
+})
+
+const getUnstarDashboardMutation = id => ({
+    type: 'delete',
+    resource: `dashboards/${id}/favorite`,
+})
+
+export const apiStarDashboard = (dataEngine, id, isStarred) => {
+    const mutation = isStarred
+        ? getStarDashboardMutation(id)
```

Comment:
Do you get console errors complaining about the mutation not being static?
This is what @amcgee suggested for FileMenu:
https://github.com/dhis2/analytics/pull/655/commits/7a54cba875b1eee79640bbc679434ba0c930c7a8

Reply:
No I'm not getting that warning.

I noticed the mutation in the link you posted had an `id` function: 

```
const mapMutation = {	
    resource: 'maps',	
    id: ({ id }) => id,	
    type: 'delete',	
}
```

Whereas star dashboard uses this: 

```
const getStarDashboardMutation = id => ({
    type: 'create',
    resource: `dashboards/${id}/favorite`,
})
```

Which doesn't have a function in it.  Not sure.

Reply:
Update: when I switched to `useDataMutation`, I started seeing the warning, even when using `useMemo`.